cohttp/test/test_request.ml

```diff
@@ -218,7 +218,7 @@ let parse_request_uri_host_query_no_slash _ =
   parse_request_uri_ r bad_request "parse_request_uri_host_query_no_slash"
 
 let parse_request_connect _ =
-  let r = "CONNECT vpn.example.net:443 HTTP/1.1\r\n" in
+  let r = "CONNECT vpn.example.net:443 HTTP/1.1\r\n\r\n" in
```

Comment:
Shouldn't CONNECT only end with a carriage return?

Reply:
I will check again, but I believe the full request does need to finish on an `\r\n` after the first CRLF that indicates the end of the first line of the request.

Reply:
No, [you are right](https://datatracker.ietf.org/doc/html/rfc7230#section-3), it is the request line (with ends with a CRLF) + CRLF

Reply:
I'm not sure how the test passed before. Maybe it was only testing the parsing of the first line. The new parser always attempts to consume the first line + all headers + CRLF